web/src/components/forms/validator-form.tsx

```diff
@@ -40,21 +45,43 @@ export const ValidatorForm: FC = () => {
   const [useExistingPEP, setUseExistingPEP] = useState(false);
   const [useExistingSchema, setUseExistingSchema] = useState(true);
   const [schemaString, setSchemaString] = useState<string | undefined>(undefined);
+  // const [activePepPasteTab, setActivePepPasteTab] = useState('pepPaste1');
 
   // watch the form data so we can use it
   const pepFiles = watch('pepFiles');
   const pepRegistryPath = watch('pepRegistryPath');
   const schemaFiles = watch('schemaFiles');
   const schemaRegistryPath = watch('schemaRegistryPath');
 
+  const [schemaPaste, setSchemaPaste] = useState<string>('');
+  // const [pep_Paste1, setPepPaste1] = useState<string>('');
+  // const [pep_Paste2, setPepPaste2] = useState<string>('');
+  
   const { data: schema } = useSchema(schemaRegistryPath?.value);
+  const params = useMemo(() => {
+    if (useExistingPEP) {
+      return {
+        pep: pepRegistryPath?.value,
+        schema: schemaString,
+        schema_registry: undefined,
+        enabled: true,
+      };
+    } else {
+      return {
+        pep: pepFiles,
+        schema: schemaString,
+        schema_registry: undefined,
+        enabled: true,
+      };
+    }
+  }, [useExistingPEP, pepFiles, pepRegistryPath?.value, schemaString]);
 
   const {
     data: result,
     error,
     isFetching: isValidating,
     refetch,
-  } = useValidation(useExistingPEP ? pepRegistryPath?.value : pepFiles, schemaString);
+  } = useValidation(params);
```

Comment:
Nice. This is clean I like this. Wonder if we should change our other mutations/queries to a similar style 🤔

Reply:
That would definitely be nice